Cesium.Compiler/FileSystemIncludeContext.cs

```diff
@@ -6,28 +6,39 @@ public class FileSystemIncludeContext : IIncludeContext
 {
     private readonly string _stdLibDirectory;
     private readonly string _currentDirectory;
+    private readonly List<string> visitedFiles = new();
 
     public FileSystemIncludeContext(string stdLibDirectory, string currentDirectory)
     {
         _stdLibDirectory = stdLibDirectory;
         _currentDirectory = currentDirectory;
     }
 
-    public ValueTask<TextReader> LookUpAngleBracedIncludeFile(string filePath) => new(Task.Run(
-        () =>
-        {
-            var path = Path.Combine(_stdLibDirectory, filePath);
-            return (TextReader)new StreamReader(path);
-        }));
-
-    public ValueTask<TextReader> LookUpQuotedIncludeFile(string filePath) => new(Task.Run(
-        () =>
-        {
-            var path = Path.Combine(_currentDirectory, filePath);
-            if (File.Exists(path))
-                return (TextReader)new StreamReader(path);
-
-            path = Path.Combine(_stdLibDirectory, filePath);
-            return (TextReader)new StreamReader(path);
-        }));
+    public string LookUpAngleBracedIncludeFile(string filePath)
+    {
+        var path = Path.Combine(_stdLibDirectory, filePath);
+        return path;
+    }
+
+    public string LookUpQuotedIncludeFile(string filePath)
+    {
+        var path = Path.Combine(_currentDirectory, filePath);
+        if (File.Exists(path))
+            return Path.GetFullPath(path);
+
+        path = Path.Combine(_stdLibDirectory, filePath);
+        return Path.GetFullPath(path);
+    }
+
+    public TextReader OpenFileStream(string filePath) => new StreamReader(filePath);
+
+    public bool CanIncludeFile(string filePath)
+    {
+        return !visitedFiles.Contains(filePath);
+    }
+
+    public void RegisterPragmaOnceFile(string filePath)
```

Comment:
_(Minor)_ I propose to adjust the names here. Let's call the `#pragma once` feature an "internal include guard", and then:
```suggestion
    public bool ShouldIncludeFile(string filePath)
    {
        return !visitedFiles.Contains(filePath);
    }

    public void RegisterGuardedFileInclude(string filePath)
```